modules/carto/src/layers/carto-dynamic-tile-layer.js

```diff
@@ -0,0 +1,87 @@
+/* global TextDecoder */
+import {log} from '@deck.gl/core';
+import {ClipExtension} from '@deck.gl/extensions';
+import {MVTLayer} from '@deck.gl/geo-layers';
+import {GeoJsonLayer} from '@deck.gl/layers';
+import {geojsonToBinary} from '@loaders.gl/gis';
+import {encodeParameter} from '../api/maps-api-common';
+import {getURLFromTemplate} from '@deck.gl/geo-layers/tile-layer/utils';
+
+function parseJSON(arrayBuffer) {
+  return JSON.parse(new TextDecoder().decode(arrayBuffer));
+}
+
+function parseCartoDynamicTile(arrayBuffer, options) {
+  if (!arrayBuffer) return null;
+  const formatTiles = options && options.cartoDynamicTile && options.cartoDynamicTile.formatTiles;
+  if (formatTiles === 'geojson') return geojsonToBinary(parseJSON(arrayBuffer).features);
+  return null;
+}
+
+const CartoDynamicTileLoader = {
+  name: 'CARTO Dynamic Tile',
+  id: 'cartoDynamicTile',
+  module: 'carto',
+  extensions: ['pbf'],
+  mimeTypes: ['application/x-protobuf'],
+  category: 'geometry',
+  worker: false,
+  parse: async (arrayBuffer, options) => parseCartoDynamicTile(arrayBuffer, options),
+  parseSync: parseCartoDynamicTile,
+  options: {
+    cartoDynamicTile: {
+      formatTiles: 'geojson'
+    }
+  }
+};
+
+export default class CartoDynamicTileLayer extends MVTLayer {
+  getTileData(tile) {
+    let url = getURLFromTemplate(this.state.data, tile);
+    if (!url) {
+      return Promise.reject('Invalid URL');
+    }
+
+    let loadOptions = this.getLoadOptions();
+    const {fetch, formatTiles} = this.props;
+    const {signal} = tile;
+
+    loadOptions = {
+      ...loadOptions,
+      mimeType: 'application/x-protobuf'
+    };
+
+    if (formatTiles) {
+      log.assert(
+        ['geojson'].includes(formatTiles),
+        `Invalid value for formatTiles: ${formatTiles}`
+      );
+      url += `&${encodeParameter('formatTiles', formatTiles)}`;
+      loadOptions.cartoBinaryTile = {formatTiles};
+    }
+
+    return fetch(url, {propName: 'data', layer: this, loadOptions, signal});
+  }
+
+  renderSubLayers(props) {
+    if (props.data === null) {
+      return null;
+    }
+
+    props.autoHighlight = false;
```

Comment:
Why are disabling autoHighlight? The backend is returning the properties as we're using the geojson

Reply:
I based the method `renderLayers` method on `MVTLayer` which does the same: https://github.com/visgl/deck.gl/blob/433c82367009e123fb2866fd6622d15ef7e4335e/modules/geo-layers/src/mvt-layer/mvt-layer.js#L144

Reply:
True, it's because mvt manage highlight on their own via https://github.com/visgl/deck.gl/blob/433c82367009e123fb2866fd6622d15ef7e4335e/modules/geo-layers/src/mvt-layer/mvt-layer.js#L162